File: app/src/main/java/com/prox/voicechanger/repository/RepositoryModule.java
package com.prox.voicechanger.repository;

import com.prox.voicechanger.database.FileVoiceDAO;

import javax.inject.Inject;
import javax.inject.Singleton;

import dagger.Module;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

@Module
@InstallIn(SingletonComponent.class)
public class RepositoryModule {

    @Inject
    @Singleton
    public FileVoiceRepository provideRepository(FileVoiceDAO dao){
        return new FileVoiceRepository(dao);
    }
}

File: app/src/main/java/com/prox/voicechanger/repository/RepositoryModule.kt
package com.prox.voicechanger.repository

import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Inject
import javax.inject.Singleton
import com.prox.voicechanger.database.FileVoiceDAO
import dagger.Module

@Module
@InstallIn(SingletonComponent::class)
class RepositoryModule {
    @Inject
    @Singleton
    fun provideRepository(dao: FileVoiceDAO): FileVoiceRepository {
        return FileVoiceRepository(dao)
    }
}